proto/sagittarius/datatype.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+import "translations.proto";
+import "google/protobuf/any.proto";
+
+option ruby_package = "Tucana::Sagittarius";
+
+package sagittarius;
+
+message DataType {
+  enum Variant {
+    UNKNOWN = 0;
+    PRIMITIVE = 1;
+    TYPE = 2;
+    OBJECT = 3;
+    DATATYPE = 4;
+    ARRAY = 5;
+    GENERIC = 6;
+    FUNCTION = 7;
+  }
+
+  shared.Translation name = 1;
```

Comment:
Should we use a `map<string, string>` instead of a repeated object?

Reply:
I think an object is more readable and easier to understand